python/ucl.c

```diff
@@ -0,0 +1,136 @@
+// Attempts to load a UCL structure from a string
+#include <ucl.h>
+#include <Python.h>
+
+static PyObject*
+_basic_ucl_type(ucl_object_t const * const obj) {
+	if (obj->type == UCL_INT) {
+		return Py_BuildValue("L", (long long)ucl_object_toint (obj));
+	}
+	else if (obj->type == UCL_FLOAT) {
+		return Py_BuildValue("d", ucl_object_todouble (obj));
+	}
+	else if (obj->type == UCL_STRING) {
+		return Py_BuildValue("s", ucl_object_tostring (obj));
+	}
+	else if (obj->type == UCL_BOOLEAN) {
+		// maybe used 'p' here?
+		return Py_BuildValue("s", ucl_object_tostring_forced (obj));
+	}
+	else if (obj->type == UCL_TIME) {
+		return Py_BuildValue("d", ucl_object_todouble (obj));
+	}
+	return NULL;
+}
+
+static PyObject*
+_iterate_valid_ucl(ucl_object_t const * obj) {
+	const ucl_object_t *tmp;
+	ucl_object_iter_t it = NULL;
+
+	tmp = obj;
+
+	while ((obj = ucl_iterate_object (tmp, &it, false))) {
```

Comment:
I'd recommend to use safe iterators here since they have a better understanding of implicit arrays.